IndirectFieldDecl

In according to transpiling file [`sqlite/shell.c`](https://github.com/Konstantin8105/c2go-rating/blob/master/sqlite/shell.c) of sqlite.
We have error:
```
panic: unknown node type: 'IndirectFieldDecl 0x2be19a8 <line:167:25> col:25 implicit fpstate 'struct _fpstate *''
```

--- a/ast/ast.go
+++ b/ast/ast.go
@@ -95,6 +95,8 @@ func Position(node Node) string {
 		return n.Position
 	case *IncompleteArrayType:
 		return ""
+	case *IndirectFieldDecl:
+		return n.Position
 	case *InitListExpr:
 		return n.Position
 	case *IntegerLiteral:
@@ -308,6 +310,8 @@ func Parse(line string) Node {
 		return parseImplicitValueInitExpr(line)
 	case "IncompleteArrayType":
 		return parseIncompleteArrayType(line)
+	case "IndirectFieldDecl":
+		return parseIndirectFieldDecl(line)
 	case "InitListExpr":
 		return parseInitListExpr(line)
 	case "IntegerLiteral":

--- a/ast/indirect_field_decl.go
+++ b/ast/indirect_field_decl.go
@@ -1,0 +1,40 @@
+package ast
+
+import "strings"
+
+type IndirectFieldDecl struct {
+	Address   string
+	Position  string
+	Position2 string
+	Implicit  bool
+	Name      string
+	Type      string
+	Children  []Node
+}
+
+func parseIndirectFieldDecl(line string) *IndirectFieldDecl {
+	groups := groupsFromRegex(
+		`<(?P<position>.*)>
+		(?P<position2> [^ ]+:[\d:]+)?
+		(?P<implicit> implicit)?
+		 (?P<name>\w+)
+		 '(?P<type>.+)'`,
+		line,
+	)
+
+	return &IndirectFieldDecl{
+		Address:   groups["address"],
+		Position:  groups["position"],
+		Position2: strings.TrimSpace(groups["position2"]),
+		Implicit:  len(groups["implicit"]) > 0,
+		Name:      groups["name"],
+		Type:      groups["type"],
+		Children:  []Node{},
+	}
+}
+
+// AddChild adds a new child node. Child nodes can then be accessed with the
+// Children attribute.
+func (n *IndirectFieldDecl) AddChild(node Node) {
+	n.Children = append(n.Children, node)
+}

--- a/ast/indirect_field_decl_test.go
+++ b/ast/indirect_field_decl_test.go
@@ -1,0 +1,21 @@
+package ast
+
+import (
+	"testing"
+)
+
+func TestIndirectFieldDecl(t *testing.T) {
+	nodes := map[string]Node{
+		`0x2be19a8 <line:167:25> col:25 implicit fpstate 'struct _fpstate *'`: &IndirectFieldDecl{
+			Address:   "0x2be19a8",
+			Position:  "line:167:25",
+			Position2: "col:25",
+			Implicit:  true,
+			Name:      "fpstate",
+			Type:      "struct _fpstate *",
+			Children:  []Node{},
+		},
+	}
+
+	runNodeTests(t, nodes)
+}

